loadBatchMessages Fails for Invalid Signature only on Android

On a specific topic, I get Error: Call to function 'XMTP.loadBatchMessages' has been rejected. → Caused by: org.xmtp.android.library.XMTPException: Invalid signature
Checking out the Kotlin code, it fails here: https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/messages/MessageV2.kt#L98
Two things
this syncs well on iOS, any idea how this would not fail on iOS but fail on Android? Equivalent Swift code: https://github.com/xmtp/xmtp-ios/blob/main/Sources/XMTP/Messages/MessageV2.swift#L47-L49
do you think the whole loadBatchMessages should fail in that case? Or should it just return the messages that work well (and notify there were some issues)
[MessageV2.kt](https://github.com/xmtp/xmtp-android/blob/main/library/src/main/java/org/xmtp/android/library/messages/MessageV2.kt)
<https://github.com/[xmtp/xmtp-android](https://github.com/xmtp/xmtp-android)|xmtp/xmtp-android>xmtp/xmtp-android | Added by https://xmtp-labs.slack.com/services/B05RXSHK0SX
[MessageV2.swift](https://github.com/xmtp/xmtp-ios/blob/main/Sources/XMTP/Messages/MessageV2.swift)
<https://github.com/[xmtp/xmtp-ios](https://github.com/xmtp/xmtp-ios)|xmtp/xmtp-ios>xmtp/xmtp-ios | Added by https://xmtp-labs.slack.com/services/B05RXSHK0SX

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -95,7 +95,7 @@ repositories {
 dependencies {
   implementation project(':expo-modules-core')
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${getKotlinVersion()}"
-  implementation "org.xmtp:android:0.7.1"
+  implementation "org.xmtp:android:0.7.3"
   implementation 'com.google.code.gson:gson:2.10.1'
   implementation 'com.facebook.react:react-native:0.71.3'
   implementation "com.daveanthonythomas.moshipack:moshipack:1.0.1"

--- a/ios/XMTPReactNative.podspec
+++ b/ios/XMTPReactNative.podspec
@@ -26,5 +26,5 @@ Pod::Spec.new do |s|
   s.source_files = "**/*.{h,m,swift}"
   s.dependency 'secp256k1.swift'
 	s.dependency "MessagePacker"
-  s.dependency "XMTP", "= 0.7.4-alpha0"
+  s.dependency "XMTP", "= 0.7.5-alpha0"
 end

